src/nodelet/adding_image_nodelet.cpp

```diff
@@ -0,0 +1,155 @@
+ *********************************************************************/
+
+// https://github.com/opencv/opencv/tree/2.4/samples/cpp/tutorial_code/ImgProc/AddingImage.cpp
+/**
+ * This is a demo of adding image (linear blending).
+ */
+
+#include <opencv2/opencv.hpp>
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/image_encodings.h>
+#include <dynamic_reconfigure/server.h>
+
+#include <image_transport/image_transport.h>
+#include <image_transport/subscriber_filter.h>
+
+#include <message_filters/subscriber.h>
+#include <message_filters/synchronizer.h>
+#include <message_filters/sync_policies/exact_time.h>
+#include <message_filters/sync_policies/approximate_time.h>
+
+#include "opencv_apps/AddingImageConfig.h"
+#include "opencv_apps/nodelet.h"
+
+namespace adding_image {
+  class AddingImageNodelet : public opencv_apps::Nodelet {
+  private:
+    boost::shared_ptr<image_transport::ImageTransport> it_;
+
+    typedef message_filters::sync_policies::ExactTime<
+      sensor_msgs::Image, sensor_msgs::Image> SyncPolicy;
+    typedef message_filters::sync_policies::ApproximateTime<
+      sensor_msgs::Image, sensor_msgs::Image> ApproxSyncPolicy;
+
+    ////////////////////////////////////////////////////////
+    // Dynamic Reconfigure
+    ////////////////////////////////////////////////////////
+    typedef adding_image::AddingImageConfig Config;
+    typedef dynamic_reconfigure::Server<Config> ReconfigureServer;
+    Config config_;
+    boost::shared_ptr<ReconfigureServer> reconfigure_server_;
+
+    image_transport::SubscriberFilter sub_image1_, sub_image2_;
+    image_transport::Publisher img_pub_;
+    boost::shared_ptr<message_filters::Synchronizer<SyncPolicy> > sync_;
+    boost::shared_ptr<message_filters::Synchronizer<ApproxSyncPolicy> > async_;
+    boost::mutex mutex_;
+
+    bool approximate_sync_;
+    double alpha_;
+    double beta_;
+    double gamma_;
+
+    void subscribe() {
+      NODELET_DEBUG("Subscribing to image topic.");
+      sub_image1_.subscribe(*it_, "image1", 1);
+      sub_image2_.subscribe(*it_, "image2", 1);
+      if (approximate_sync_) {
+        async_ =
+          boost::make_shared<message_filters::Synchronizer<ApproxSyncPolicy> >(
+            100);
+        async_->connectInput(sub_image1_, sub_image2_);
+        async_->registerCallback(
+          boost::bind(&AddingImageNodelet::add, this, _1, _2));
+      } else {
+        sync_ =
+          boost::make_shared<message_filters::Synchronizer<SyncPolicy> >(100);
+        sync_->connectInput(sub_image1_, sub_image2_);
+        sync_->registerCallback(
+          boost::bind(&AddingImageNodelet::add, this, _1, _2));
+      }
+    }
+
+    void unsubscribe() {
+      sub_image1_.unsubscribe();
+      sub_image2_.unsubscribe();
+    }
+
+    void reconfigureCallback(Config& config, uint32_t level) {
+      boost::mutex::scoped_lock lock(mutex_);
+      config_ = config;
+      alpha_ = config.alpha;
+      beta_ = config.beta;
```

Comment:
why you set alpha and beta individually,test test
code calculate beta from alpha, 
https://github.com/opencv/opencv/blob/2.4/samples/cpp/tutorial_code/ImgProc/AddingImages.cpp#L43